File: netconsole/models/api/task.py
from __future__ import annotations

from typing import Any

from pydantic import Field

from netconsole.models.api.common import ApiModel
from netconsole.models.task_state import TaskState


class TaskDTO(ApiModel):
    id: str
    type: str
    name: str
    status: TaskState
    progress: int = 0
    stage: str = ""
    current: int = 0
    total: int = 0
    message: str = ""
    created_time: str
    started_time: str = ""
    finished_time: str = ""
    updated_time: str
    owner: str = ""
    device: str = ""
    agent: str = ""
    result_path: str = ""
    error_message: str = ""
    result: dict[str, Any] = Field(default_factory=dict)
    source: str = "local"
    cancellable: bool = False


class TaskEventDTO(ApiModel):
    sequence: int = 0
    id: str
    task_id: str
    type: str
    time: str
    source: str = "service"
    payload: dict[str, Any] = Field(default_factory=dict)


class TaskCancelResponse(ApiModel):
    id: str
    status: TaskState
    message: str

File: apps/web/src/types/task.ts
export type TaskStatus =
  | 'PENDING'
  | 'STARTING'
  | 'RUNNING'
  | 'STOPPING'
  | 'COMPLETED'
  | 'FAILED'
  | 'CANCELLED'
  | 'CREATED'
  | 'QUEUED'
  | 'ABORTED'
  | 'STOPPED'
  | 'WARNING'
  | 'UNKNOWN'

export interface TaskItem {
  id: string
  type: string
  name: string
  status: TaskStatus
  progress: number
  phase: string
  stage: string
  message: string
  site_name: string
  owner: string
  executor: string
  source: string
  device_id: string
  device_name: string
  agent: string
  mr_name: string
  session_id: string
  mapping_state: string
  created_time: string
  started_time: string
  finished_time: string
  updated_time: string
  duration_seconds: number
  error_code: string
  error_summary: string
  has_warning: boolean
  text_integrity?: 'ok' | 'historical_corrupted' | 'current_corrupted' | 'unknown_corrupted'
  text_integrity_reason?: string
  text_integrity_updated_at?: string
  text_schema_version?: number
  producer_kind?: 'local_worker' | 'local_backend' | 'agent' | 'imported' | 'legacy'
  producer_version?: string
  producer_commit?: string
  snapshot_id: number | null
  records_count: number | null
  parser_version: string
  module?: string
  cancellable?: boolean
  cancel_reason?: string
  retryable?: boolean
  retry_reason?: string
  artifact_download?: {
    artifact_id: string
    display_name: string
    size_bytes: number
    media_type: string
    api_path: string
    query: Record<string, string>
  } | null
  artifact_reason?: string
  details?: Record<string, unknown>
}

export interface TaskLogLine {
  sequence: number
  time: string
  level: string
  type: string
  source: string
  message: string
  details?: Record<string, unknown>
}

export interface TaskLogTail {
  task_id: string
  lines: TaskLogLine[]
  message: string
}

export interface TaskSummary {
  total: number
  active: number
  completed: number
  failed: number
  warning: number
}
